workspace/viewer/PdfJsViewer.ts

```diff
@@ -478,29 +480,32 @@ export class PdfJsViewer {
       canvas.removeClass("shadow");
     }
     const canvasOffset = canvas.position();
-    const textLayer = $(".textLayer", this.jqRoot);
+    const textLayer = parent.find(".textLayer");
     textLayer.css({
       top : canvasOffset.top,
       left : canvasOffset.left
     });
   }
 
   resetCanvas() {
-    if (this.pdfPageView) {
-      this.pdfPageView.destroy();
+    if (this.loadedPages.length != 0) {
+      for(let page of this.loadedPages){
+        page.destroy();
+      }
+      this.loadedPages = [];
     }
-    this.pdfPageView = undefined;
     this.jqRoot.empty();
     this.queue.clear();
   }
 
-  private openCurrentPage() {
-    if (this.currentFileUrl) {
-      let lastCompletedMainFileId = (this.queue.isEmpty() && this.queue.lastCompleted)
-          ? this.queue.lastCompleted.mainFileId : undefined;
-      this.show(this.currentFileUrl, this.currentPage, true, lastCompletedMainFileId);
+  public showPage(page: number) {
+    if(page > this.numPages){
+      page = this.numPages;
     }
+    let pageWrapper = $("div").find(`[data-page-number='${page}']`)[0];
```

Comment:
You just search in all divs on the page? This looks like pretty broad target. Also, is anyone defining those `data-page-number` attrs?

Reply:
Done.